continuous_training/composer/labs/chicago_taxi_dag.py

```diff
@@ -0,0 +1,465 @@
+        Variable.set("NEW_VERSION_NAME",
+                     "v_{0}"
+                     .format(datetime.datetime.now().strftime("%Y%m%d%H%M%S")))
+
+    # TODO 3: Fill in arguments for python_new_version_name_op
+    python_new_version_name_op = PythonOperator(
+        provide_context=True,
+        #ADD YOUR CODE HERE
+        )
+
+    # Arguments for MLEngineTrainingOperator
+    job_id = "chicago_{}_{}".format(model.replace(".", "_"),
+                                    datetime.datetime.now()
+                                    .strftime("%Y%m%d%H%M%S"))
+    output_dir = (BUCKET + "/chicago/trained_model/{}"
+                  .format(model.replace(".", "_")))
+    log_dir = (BUCKET + "/chicago/training_logs/{}"
+                  .format(model.replace(".", "_")))
+    job_dir = JOB_DIR + "/" + job_id
+
+    # Arguments expected by the trainer package.
+    training_args = [
+        "--job-dir", job_dir,
+        "--output_dir", output_dir,
+        "--log_dir", log_dir,
+        "--train_data_path", train_files + "chicago_taxi_trips/*.csv",
+        "--eval_data_path", valid_files + "chicago_taxi_trips/*.csv",
+        "--output_ds", f"{PROJECT_ID}.{DESTINATION_DATASET}",
+        "--version_name", Variable.get('NEW_VERSION_NAME'),
+    ]
+
+    # BashOperator to remove the old SavedModel to ensure we are doing a
+    # fresh run.
+    bash_remove_trained_model_op = BashOperator(
+        task_id="bash_remove_old_trained_model_{}_task"
+                .format(model.replace(".", "_")),
+        bash_command=("if gsutil ls {0} 2> /dev/null;"
+                      "then gsutil -m rm -rf {0}/*; else true; fi"
+                      .format(output_dir + model.replace(".", "_"))),
+        dag=dag)
+
+    # Task to submit AI Platform training job
+    ml_engine_training_op = MLEngineTrainingOperator(
+        task_id="ml_engine_training_{}_task".format(model.replace(".", "_")),
+        project_id=PROJECT_ID,
+        job_id=job_id,
+        package_uris=[PACKAGE_URI],
+        training_python_module="trainer.task",
+        training_args=training_args,
+        region=REGION,
+        scale_tier="BASIC",
+        runtime_version="2.1",
+        python_version="3.7",
+        dag=dag
+     )
+
+    # SQL Query to check if new model metrics satisfy conditions. Condition is
+    # set to be high to ensure that the model likely meets the conditions.
+    model_check_sql = """
+                SELECT
+                    IF(rmse - 10.0 >= 0, rmse - 10.0, 0) AS rmse
+                FROM
+                    `{0}.{1}.{2}`
+                WHERE
+                    version_name = '{3}'
+                """.format(PROJECT_ID, DESTINATION_DATASET, 'model_metrics',
+                           Variable.get("NEW_VERSION_NAME"))
+
+    # TODO 4: Create bq_check_rmse_query_op
+    bq_check_rmse_query_op = BigQueryValueCheckOperator(
+        tolerence=0,
+        use_legacy_sql=False,
+        #ADD YOUR CODE HERE
```

Comment:
Will you speak to pass_value argument? You could consider supplying that parameter value as well.

Reply:
The pass_value argument is addressed within the lab instructions, so I didn't do that here. Part of the TODO for the students will be to include that argument with the proper value.

Reply:
SGTM